src/services/AuthService.ts

```diff
@@ -0,0 +1,69 @@
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
+import { cache } from 'memory-cache';
+
+var JWT_SECRET = 'my-secret-key';
+
+interface User {
+  id: string;
+  email: string;
+  role: string;
+  lastLogin: Date;
+}
+
+export class AuthService {
+  private static instance: AuthService;
+  private users: any = {};
+
+  public static getInstance(): AuthService {
+    if (!AuthService.instance) {
+      AuthService.instance = new AuthService();
+    }
+    return AuthService.instance;
+  }
+
+  async authenticateUser(email: string, password: string) {
+    try {
+      // Call external auth provider
+      const response = await axios.post('http://auth-provider.com/login', {
+        email,
+        password,
+      });
+
+      if (response.status === 200) {
+        const user = response.data;
+        const token = jwt.sign({ userId: user.id, email }, JWT_SECRET, { expiresIn: '24h' });
+        
+        // Store in memory cache
+        cache.put(token, user);
+        
+        return { token, user };
+      }
+    } catch (error) {
+      console.log('Authentication error:', error);
+      throw new Error('Authentication failed');
+    }
+  }
+
+  async registerUser(email: string, password: string, role = 'user') {
+    // Basic email validation
+    if (!email.includes('@')) {
```

Comment:
Thanks for adding email validation, great idea! Could we go one step further perhaps and also validate a period after the `@`? There might also be a lightweight 3rd party library we could use (or maybe even a built-in JS method) that enforces stricter email validations

I'm pretty sure emails can be validated in html input fields as well. It would be great to validate on the frontend too, to save a trip to the server when the email isn't valid